File: RepoTransBench/target_projects/Java/Go/markzhai_DataBindingAdapter/tests/original/singletype_adapter_test.go
package original

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func setupSingleTypeAdapter(layoutRes int) *SingleTypeAdapter[string] {
	return NewSingleTypeAdapter[string](layoutRes)
}

func TestSingleTypeAdapter_ConstructorAndGetters(t *testing.T) {
	adapter := setupSingleTypeAdapter(123)
	assert.Equal(t, 0, adapter.GetItemCount())
	assert.Equal(t, 123, adapter.GetLayoutRes())
}

func TestSingleTypeAdapter_Add(t *testing.T) {
	adapter := setupSingleTypeAdapter(321)
	adapter.Add("foo")
	assert.Equal(t, 1, adapter.GetItemCount())
}

func TestSingleTypeAdapter_AddAtPosition(t *testing.T) {
	adapter := setupSingleTypeAdapter(321)
	adapter.Add("foo")
	adapter.AddAt(0, "bar")
	assert.Equal(t, 2, adapter.GetItemCount())
}

func TestSingleTypeAdapter_SetAndAddAll(t *testing.T) {
	adapter := setupSingleTypeAdapter(321)
	adapter.Set([]string{"a", "b"})
	assert.Equal(t, 2, adapter.GetItemCount())
	adapter.AddAll([]string{"c"})
	assert.Equal(t, 3, adapter.GetItemCount())
}

File: RepoTransBench/target_projects/Java/C#/markzhai_DataBindingAdapter/tests/original/SingleTypeAdapterTest.cs
using System.Collections.Generic;
using Xunit;
using Moq;
using MarkZhai.DataBindingAdapter;

namespace MarkZhai.DataBindingAdapter.Tests.Original
{
    public class SingleTypeAdapterTest
    {
        private object context;
        private object inflater;

        public SingleTypeAdapterTest()
        {
            context = new object();
            inflater = new object();
        }

        [Fact]
        public void TestConstructorAndGetters()
        {
            var adapter = new SingleTypeAdapter<string>(context, 123);
            Assert.Equal(0, adapter.GetItemCount());
            Assert.Equal(123, adapter.GetLayoutRes());
        }

        [Fact]
        public void TestAdd()
        {
            var adapter = new SingleTypeAdapter<string>(context, 321);
            adapter.Add("foo");
            Assert.Equal(1, adapter.GetItemCount());
        }

        [Fact]
        public void TestAddAtPosition()
        {
            var adapter = new SingleTypeAdapter<string>(context, 321);
            adapter.Add("foo");
            adapter.Add(0, "bar");
            Assert.Equal(2, adapter.GetItemCount());
        }

        [Fact]
        public void TestSetAndAddAll()
        {
            var adapter = new SingleTypeAdapter<string>(context, 321);
            adapter.Set(new List<string> { "a", "b" });
            Assert.Equal(2, adapter.GetItemCount());
            adapter.AddAll(new List<string> { "c" });
            Assert.Equal(3, adapter.GetItemCount());
        }
    }
}